### PR TITLE
Python 3

### PR DESCRIPTION
**Pull request by nim65s**
Follows #31

### CODE REVIEW

**Comment by nim65s** [comment]
(this still requires https://github.com/jrl-umi3218/jrl-cmakemodules/pull/241)

File: src/dynamic_graph/matlab.py [hunk 1]
```diff
@@ -4,104 +4,103 @@
 # using the str() member.
```
[the remark this answers is not in the document]
Comment (nim65s) [reply src/dynamic_graph/matlab.py]: I have no idea, and I don't know if this is used anywhere…

[the remark this answers is not in the document]
Comment (MaximilienNaveau) [reply src/dynamic_graph/matlab.py]: @olivier-stasse do you have an idea about this?

[the remark this answers is not in the document]
Comment (jmirabel) [reply src/dynamic_graph/matlab.py]: It should be removed.

[the remark this answers is not in the document]
Comment (MaximilienNaveau) [reply src/dynamic_graph/matlab.py]: thanks @jmirabel !
@nim65s can you deal with that? And then we should merge this pull request.

[the remark this answers is not in the document]
Comment (nim65s) [reply src/dynamic_graph/matlab.py]: I'll remove it, but please don't merge, I'll open more a PR for all dependent package in the SoT, and I will merge all of those at once, with a release.

File: src/dynamic_graph/dynamic-graph-py.cc [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  }
+  if (!PyCapsule_CheckExact(objIn)) {
+    PyErr_SetString(PyExc_TypeError,
+                    "second argument should be a pointer to"
+                    " signalBase<int>.");
+    return NULL;
+  }
+
+  pObjIn = PyCapsule_GetPointer(objIn, "dynamic_graph.Signal");
+  SignalBase<int>* signalIn = (SignalBase<int>*)pObjIn;
```
Comment (jmirabel) [inline src/dynamic_graph/dynamic-graph-py.cc]: Could you add 

    if (signalIn == NULL) return NULL;

With the current release, one get a very annoying SEGV when passing the wrong object.

File: src/dynamic_graph/dynamic-graph-py.cc [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  pObjOut = PyCapsule_GetPointer(objOut, "dynamic_graph.Signal");
+  SignalBase<int>* signalOut = (SignalBase<int>*)pObjOut;
```
Comment (jmirabel) [inline src/dynamic_graph/dynamic-graph-py.cc]: same as above.

File: src/dynamic_graph/dynamic-graph-py.cc (hunk 2 above)
Comment (jmirabel) [reply src/dynamic_graph/dynamic-graph-py.cc]: PyCapsule_GetPointer is responsible for setting the Python error. No need to handle it.

Comment (nim65s) [reply src/dynamic_graph/dynamic-graph-py.cc]: I should do that, thanks for the review. I'll also provide a test to check what happens if we pass a wrong object.

**Pull request merged**